client/web/compose/src/mixins/pages.js

```diff
@@ -0,0 +1,56 @@
+import { compose, NoID } from '@cortezaproject/corteza-js'
+
+export default {
+  methods: {
+    countDuplicatePages (pages, pattern) {
+      let substrTitle = this.page.title
+
+      if (this.page.title.includes(this.$t('copyOf')) && !pattern.test(this.page.title)) {
+        substrTitle = this.page.title.substring(this.$t('copyOf').length, this.page.title.length)
+      } else if (this.page.title.includes(this.$t('copyOf')) && pattern.test(this.page.title)) {
+        substrTitle = this.page.title.substring(this.$t('copyOf').length, this.page.title.lastIndexOf('(') - 1)
+      }
+
+      return pages.filter(p => p.title.includes(substrTitle)).length
+    },
+
+    async handleClone (name) {
+      try {
+        const { namespaceID = NoID } = this.namespace
+        let page = this.page.clone()
+        // find a number inside parentheses
+        const pattern = /\([0-9]+\)/
+
+        const pages = await this.loadPages({ namespaceID, force: true })
+
+        let newPageTitle = this.$t('copyOf', { title: this.page.title })
+        const countDuplicates = this.countDuplicatePages(pages, pattern)
+
+        if (countDuplicates === 2) {
+          newPageTitle = `${this.page.title} (${countDuplicates})`
+        } else if (countDuplicates > 2 && this.page.title.includes(this.$t('copyOf'))) {
```

Comment:
Add comments here aswell and maybe think about rewriting the condition so that `countDuplicates > 2` doesn't repeat